python/pyspark/mllib/linalg/__init__.py

```diff
@@ -794,6 +794,13 @@ def __eq__(self, other):
             return Vectors._equals(self.indices, self.values, list(xrange(len(other))), other.array)
         return False
 
+    def __getattr__(self, item):        
+        if _have_scipy:
+            csr = scipy.sparse.csr_matrix((self.values, self.indices, [0, 2]))
+            return getattr(csr, item)
+        else:
+            return self
```

Comment:
Systems without Scipy return the following: 
```
from pyspark.mllib.linalg import SparseVector
a = SparseVector(4, {1: 1.0, 3: 5.5})
a.sum()

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'SparseVector' object is not callable
```
This has been corrected in the latest commit.
 `__getattr__` essentially is the catch for "any attribute that does not have a name" so standard behavior should be an AttributeError not just returning the object (thank you for catching this!). The new code gives the following error when SciPy is not available:

```
a.sum()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/sobh/spark/python/pyspark/mllib/linalg/__init__.py", line 802, in __getattr__
    raise AttributeError("'{0}' object has no attribute '{1}'.".format(self.__class__, item))
AttributeError: '<class 'pyspark.mllib.linalg.SparseVector'>' object has no attribute 'sum'.


```
-----------
Systems with Scipy will give the result as expected:
```
from pyspark.mllib.linalg import SparseVector
a = SparseVector(4, {1: 1.0, 3: 5.5})
a.sum()

6.5
```